src/lib/mongoBackend/mongoQueryTypes.cpp

```diff
@@ -236,14 +236,38 @@ HttpStatusCode mongoEntityTypes
    *
    * However, considering that the number of types will be small compared with the number of entities,
    * the current approach seems to be ok
+   *
+   * emptyEntityType is special: it must aggregate results for entity type "" and for entities without type.
+   * Is pre-created before starting processing results and destroyed if at the end it has not been used
+   * (i.e. pushed back into the vector)
+   *
    */
+
+  EntityType* emptyEntityType     = new EntityType("");
+  bool        emptyEntityTypeUsed = false;
+
   for (unsigned int ix = offset; ix < MIN(resultsArray.size(), offset + limit); ++ix)
   {
-    BSONObj                   resultItem  = resultsArray[ix].embeddedObject();
-    EntityType*               entityType  = new EntityType(getStringField(resultItem, "_id"));
-    std::vector<BSONElement>  attrsArray  = getField(resultItem, "attrs").Array();
+    BSONObj                   resultItem  = resultsArray[ix].embeddedObject();    
+    std::vector<BSONElement>  attrsArray  = getFieldF(resultItem, "attrs").Array();
+
+
+    EntityType*               entityType;
 
-    entityType->count = countEntities(tenant, servicePathV, entityType->type);
+    bool nullId = ((resultItem.hasField("")) && (getFieldF(resultItem, "").isNull())) || (getStringFieldF(resultItem, "_id") == "");
```

Comment:
Difficult line ...
resultItem.hasField("") I imagine refers to that resultItem has a field with empty name?
Perhaps a comment line or two ...